dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/context/annotation/DubboClassPathBeanDefinitionScanner.java

```diff
@@ -67,4 +74,35 @@ public boolean checkCandidate(String beanName, BeanDefinition beanDefinition) th
         return super.checkCandidate(beanName, beanDefinition);
     }
 
+    @Override
+    public Set<BeanDefinition> findCandidateComponents(String basePackage) {
+        Set<BeanDefinition> beanDefinitions = beanDefinitionMap.get(basePackage);
+        if (CollectionUtils.isEmpty(beanDefinitions)) {
+            beanDefinitions = super.findCandidateComponents(basePackage);
+            beanDefinitionMap.put(basePackage, beanDefinitions);
+        }
+        return beanDefinitions;
+    }
+
+    /**
+     * Registers @Service Bean and Finds all BeanDefinitionHolders of @Service
+     *
+     * @param registry              BeanDefinitionRegistry
+     * @param beanNameGenerator     BeanNameGenerator
+     * @param basePackage           basePackage
+     * @return                      all BeanDefinitionHolders of @Service
+     */
+    public Set<BeanDefinitionHolder> doScan(BeanDefinitionRegistry registry, BeanNameGenerator beanNameGenerator, String basePackage) {
+        // Registers @Service Bean first
+        super.doScan(basePackage);
+        Set<BeanDefinition> beanDefinitions = findCandidateComponents(basePackage);
```

Comment:
`super.doScan(basePackage)`: will do first scanning 
`findCandidateComponents()` : will do second scanning
This did not reduce scanning.

Reply:
Please look i'm pull request.
Rewrite  method(Set<BeanDefinition> findCandidateComponents(String basePackage)) in DubboClassPathBeanDefinitionScanner.

Reply:
I rewrite method **findCandidateComponents** for DubboClassPathBeanDefinitionScanner.

Reply:
I think just cache `Set<BeanDefinitionHolder>` in  `findCandidateComponents` method is enough,  do not need add new `doScan` method